src/comments/reaction.cpp

```diff
@@ -0,0 +1,116 @@
+#include <comments/reaction.hpp>
+#include <comments/likeable.hpp>
+#include <dao.hpp>
+#include <document_graph/edge.hpp>
+
+#define TYPED_DOCUMENT_TYPE document_types::REACTION
+#define CONTENT_GROUP_LABEL_REACTION "reaction"
+#define CONTENT_REACTION_TYPE "type"
+#define CONTENT_REACTION_WHO "who"
+
+namespace hypha
+{
+    static constexpr eosio::name REACTION_LIKE("liked");
+
+
+    static bool is_reaction(eosio::name reaction)
+    {
+        return reaction == REACTION_LIKE;
+    }
+
+    Reaction::Reaction(hypha::dao& dao, uint64_t id)
+    : TypedDocument(dao, id, TYPED_DOCUMENT_TYPE)
+    {
+        TRACE_FUNCTION()
+    }
+
+    Reaction::Reaction(
+        dao&dao,
+        Likeable& likeable,
+        const eosio::name who,
+        const eosio::name reaction
+    )
+    : TypedDocument(dao, TYPED_DOCUMENT_TYPE)
+    {
+        TRACE_FUNCTION()
+
+        uint64_t memberId = dao.getMemberID(who);
+        EOS_CHECK(
+            !Edge::exists(dao.get_self(), memberId, likeable.getId(), common::REACTED),
+            "Member already reacted to this document"
+        );
+        EOS_CHECK(
+            is_reaction(reaction),
+            "Only like reaction currently supported"
+        );
+
+        ContentGroups contentGroups{
+            ContentGroup{
+                Content(CONTENT_GROUP_LABEL, CONTENT_GROUP_LABEL_REACTION),
+                Content(CONTENT_REACTION_TYPE, reaction),
+                Content(CONTENT_REACTION_WHO, who),
+            }
+        };
+
+        initializeDocument(dao, contentGroups);
+        Edge::getOrNew(dao.get_self(), who, memberId, likeable.getId(), common::REACTED);
+        Edge::getOrNew(dao.get_self(), who, likeable.getId(), memberId, common::REACTED_TO);
```

Comment:
Shouldn't this be the other way around? Also, IMHO it would be clearer if we change `reacted` to `reactedby` 
what do you think?
`member -> reactedto -> likeable`
`likeable -> reactedby -> member`